util.go

```diff
@@ -144,6 +144,20 @@ func backoff(base time.Duration, round, limit uint64) time.Duration {
 	return base
 }
 
+// cappedExponentialBackoff computes the exponential backoff with an adjustable
+// cap on the max timeout.
+func cappedExponentialBackoff(base time.Duration, round, limit uint64, cap time.Duration) time.Duration {
+	power := min(round, limit)
+	for power > 2 {
+		if base > cap {
+			return cap
+		}
+		base *= 2
+		power--
+	}
+	return base
```

Comment:
I think there is still potential for base > cap here. Probably we should move the if condition to right before the return value. 
```suggestion
	if base > cap {
		return cap
	}
	return base
```

Reply:
Makes sense!